### PR TITLE
Tabbed interface: provide option to make session storage optional

### PR DESCRIPTION
**Pull request by crichard**

### CODE REVIEW

**Comment by crichard** [comment]
Should address #6538

File: src/plugins/tabs/tabs.js [hunk 1]
```diff
@@ -457,12 +464,14 @@ var componentName = "wb-tabs",
 				.addClass( "active" );
 
 		// Update sessionStorage with the current active panel
-		try {
-			sessionStorage.setItem(
-				pagePath + $container.attr( "id" ) + activePanel,
-				$next.attr( "id" )
-			);
-		} catch ( error ) {
+		if ( !$container.data( componentName ).settings.ignoreSession ) {
```
Comment (pjackson28) [inline src/plugins/tabs/tabs.js]: The session object should be cached since it is now being used more than once.

File: src/plugins/tabs/tabs.js [hunk 2]
```diff
@@ -869,12 +878,14 @@ $document.on( activateEvent, selector + " > .tabpanels > details > summary", fun
 		$details = $( details );
 
 		// Update sessionStorage with the current active panel
-		try {
-			sessionStorage.setItem(
-				pagePath + $details.closest( selector ).attr( "id" ) + activePanel,
-				details.id
-			);
-		} catch ( error ) {
+		if ( !$container.data( componentName ).settings.ignoreSession ) {
```
Comment (pjackson28) [inline src/plugins/tabs/tabs.js]: The settings object should be cached since it is now being used more than once.

**Comment by LaurentGoderre** [comment]
Looks good to me!

**Comment by LaurentGoderre** [comment]
@crichard can you fix the indentation to use tabs instead of spaces for these two lines (perhaps you can set your editor to do that by default)

```js
$details = $( details );
tabSettings = $container.data( componentName ).settings;
```

**Comment by LaurentGoderre** [comment]
Thanks so much for the PR!

**Pull request merged**